src/renderer/api/catalog-entity-registry.ts

```diff
@@ -29,19 +29,18 @@ import { Disposer, iter } from "../utils";
 import { once } from "lodash";
 import logger from "../../common/logger";
 import { catalogEntityRunContext } from "./catalog-entity";
+import { CatalogRunEvent } from "../../common/catalog/catalog-run-event";
 
 export type EntityFilter = (entity: CatalogEntity) => any;
-export type CatalogEntityOnBeforeRun = (entity: CatalogEntity) => boolean | Promise<boolean>;
-
-type CatalogEntityUid = CatalogEntity["metadata"]["uid"];
+export type CatalogEntityOnBeforeRun = ((event: CatalogRunEvent) => void) | ((event: CatalogRunEvent) => Promise<void>);
```

Comment:
This can be simplified
```suggestion
export type CatalogEntityOnBeforeRun = (event: CatalogRunEvent) => void | Promise<void>;
```